js/core/seo.js

```diff
@@ -0,0 +1,55 @@
+
+// Module core/seo
+// Manages SEO information for documents
+// e.g. set the canonical URL for the document if configured
+
+define(
+    ["core/pubsubhub"],
+    function (pubsubhub) {
+        return {
+            run:    function (conf, doc, cb) {
+                var warnings = [];
```

Comment:
It's probably just as effective to just emit the warnings as you go, instead of collecting them.